packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -0,0 +1,444 @@
+          new Offset(
+              wedgeX, wedgeY + (calc.tooltipBelow ? -wedgeSize : wedgeSize)),
+        ], true),
+      tooltipBackground);
+
+  textPainter.paint(
+      canvas, tipOffset + const Offset(_kTooltipPadding, _kTooltipPadding));
+  canvas.restore();
+}
+
+final Color _kHighlightedRenderObjectFillColor =
+    const Color.fromARGB(128, 128, 128, 255);
+final Color _kHighlightedRenderObjectBorderColor =
+    const Color.fromARGB(128, 64, 64, 128);
+
+class _WidgetInspectorPainter extends CustomPainter {
+  const _WidgetInspectorPainter(this.owner, this.generation, this.selected);
+
+  final PipelineOwner owner;
+  final int generation;
+  final RenderBox selected;
+
+  bool get isSelectionActive => selected != null && selected.attached;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    if (!isSelectionActive)
+      return;
+
+    canvas.save();
+
+    final Matrix4 transform = selected.getTransformTo(null);
+    // Highlight the selected renderObject.
+    canvas
+      ..save()
+      ..transform(transform.storage)
+      ..drawRect(
+          (Offset.zero & selected.size).deflate(0.5),
+          new Paint()
+            ..style = PaintingStyle.fill
+            ..color = _kHighlightedRenderObjectFillColor)
+      ..drawRect(
+          (Offset.zero & selected.size).deflate(0.5),
+          new Paint()
+            ..style = PaintingStyle.stroke
+            ..strokeWidth = 1.0
+            ..color = _kHighlightedRenderObjectBorderColor)
+      ..restore();
+
+    final List<Vector3> corners = <Vector3>[
+      transform.transform3(new Vector3(0.0, 0.0, 0.0)),
+      transform.transform3(new Vector3(selected.size.width, 0.0, 0.0)),
+      transform.transform3(
+          new Vector3(selected.size.width, selected.size.height, 0.0)),
+      transform.transform3(new Vector3(0.0, selected.size.height, 0.0))
+    ];
+
+    // Compute an axis aligned bounding box for selected.
+    final double left = corners.fold<double>(
+        double.MAX_FINITE, (double p, Vector3 v) => math.min(p, v.x));
+    final double top = corners.fold<double>(
+        double.MAX_FINITE, (double p, Vector3 v) => math.min(p, v.y));
+    final double right =
+        corners.fold<double>(0.0, (double p, Vector3 v) => math.max(p, v.x));
+    final double bottom =
+        corners.fold<double>(0.0, (double p, Vector3 v) => math.max(p, v.y));
+
+    final Offset target = new Offset(left, (top + bottom) / 2);
+    final double offsetFromWidget = 9.0;
+    final double verticalOffset = (bottom - top) / 2 + offsetFromWidget;
+
+    final Rect targetRect = new Rect.fromLTRB(left, top, right, bottom);
+
+    _paintDescription(
+        canvas, selected, target, verticalOffset, size, targetRect);
+
+    // TODO(jacobr): provide an option to perform a debug paint of just the
+    // selected widget.
+
+    canvas.restore();
+  }
+
+  @override
+  bool shouldRepaint(_WidgetInspectorPainter oldDelegate) {
+    return isSelectionActive ||
```

Comment:
So in particular, for example, the paint method right now depends on "selected.getTransformTo(null)". But you can't do that, because that will change without notifying you.